config/dependency.go

```diff
@@ -330,7 +330,9 @@ func dependencyBlocksToCtyValue(ctx *ParsingContext, dependencyConfigs []Depende
 				return err
 			}
 			if dependencyConfig.RenderedOutputs != nil {
+				lock.Lock()
```

Comment:
I was wondering if should use a shared lock with `dependencyMap`

Reply:
I mean that `lock := sync.Mutex{}` defined in 318 is used to isolate access to different structures: `paths` and `dependencyMap` 
https://github.com/gruntwork-io/terragrunt/blob/c9e00e17d2ee8ef65e7a8623a2f33d6d912d164a/config/dependency.go#L334

https://github.com/gruntwork-io/terragrunt/blob/c9e00e17d2ee8ef65e7a8623a2f33d6d912d164a/config/dependency.go#L352

Race condition will occur if will be used separated lock instances? one for `paths` and different for `dependencyMap` ?